### PR TITLE
add pair layout and FixedPairGalaxyCatalog

### PR DESCRIPTION
**Pull request by esheldon**
pair sims

### CODE REVIEW

File: descwl_shear_sims/shifts.py [hunk 1]
```diff
@@ -129,3 +145,38 @@ def get_random_shifts(*, rng, dim, buff, size):
...
+    Returns
+    -------
+    shifts: array
+        Array with dx, dy offset fields for each point, in
+        arcsec
+    """
+
+    shifts = np.zeros(2, dtype=[('dx', 'f8'), ('dy', 'f8')])
+
+    angle = rng.uniform(low=0, high=2*np.pi)
```
Comment (beckermr) [inline descwl_shear_sims/shifts.py:170]: This angle should only be to pi and not 2pi.

File: descwl_shear_sims/shifts.py [hunk 2]
```diff
@@ -129,3 +145,38 @@ def get_random_shifts(*, rng, dim, buff, size):
...
+    shift_radius = sep / 2
+
+    dx = np.cos(angle)*shift_radius
+    dy = np.sin(angle)*shift_radius
+
+    shifts['dx'][0] = dx
+    shifts['dy'][0] = dy
+
+    shifts['dx'][1] = -dx
+    shifts['dy'][1] = -dy
```
Comment (beckermr) [inline descwl_shear_sims/shifts.py:180]: We should dither these positions?

File: descwl_shear_sims/shifts.py (hunk 1 above)
Comment (esheldon) [reply descwl_shear_sims/shifts.py:170]: true, but it will give the same result

File: descwl_shear_sims/shifts.py (hunk 2 above)
Comment (esheldon) [reply descwl_shear_sims/shifts.py:180]: We have not been dithering the grid, relying on dithering at the image level.  I actually have an issue on this #155

File: descwl_shear_sims/shifts.py (hunk 1 above)
Comment (beckermr) [reply descwl_shear_sims/shifts.py:170]: My student and I found weird results using 2pi but maybe that was another bug.

File: descwl_shear_sims/shifts.py (hunk 2 above)
Comment (beckermr) [reply descwl_shear_sims/shifts.py:180]: I think we want dithers in the coadd system too just be safe. It certainly can't hurt. I'd strongly suggest doing it.

Comment (esheldon) [reply descwl_shear_sims/shifts.py:180]: the shifts codes do not know about the pixel scale, so this would require some restructuring

Comment (esheldon) [reply descwl_shear_sims/shifts.py:180]: An easier option would be to require image dithers

Comment (beckermr) [reply descwl_shear_sims/shifts.py:180]: Image dithers still leave the objects at the same spot in the coadd system. This is still a potential source of bias.

Comment (esheldon) [reply descwl_shear_sims/shifts.py:180]: actually we are dithering the grid, so that issue must be out of date

**Comment by beckermr** [comment]
One more thing. I think the dithers of the two objects should be independent.

**Comment by esheldon** [comment]
how big?  A pixel is significant for these object sizes

**Comment by esheldon** [comment]
I kept sep same because a pixel sized dither would dramatically change the blending

**Comment by beckermr** [comment]
Hmmmm ok. Let's keep it as is then. Thanks for the back and forth here!

**Pull request merged**